backend/adumbra/database/images.py

```diff
@@ -89,8 +54,16 @@ class ImageModel(ShimmedDynamicDocument):
     events = fields.EmbeddedDocumentListField(Event)
     regenerate_thumbnail = fields.BooleanField(default=False)
 
+    # TODO: determine how to accomplish this without overriding the delete method
+    def delete(self, *args, **kwargs):
+        delete_thumbnail(self.path)
+        AnnotationModel.objects(image_id=self.id).delete()
+        return super(ImageModel, self).delete(*args, **kwargs)
+
     @classmethod
-    def create_from_path(cls, path, dataset_id=None):
+    def create_from_path(cls, path, dataset_id):
```

Comment:
Can you add type hints?

(I think dataset_id is an int?)
```suggestion
    def create_from_path(cls, path: str, dataset_id: int):
```

Reply:
@jdrakes is this resolved with your todo comment? OK to merge?

Reply:
Yes